File: BE/src/main/java/cz/osu/opr3/spaceshipregister/services/components/ComponentProducerService.java
package cz.osu.opr3.spaceshipregister.services.components;

import cz.osu.opr3.spaceshipregister.models.entities.components.ComponentProducer;
import cz.osu.opr3.spaceshipregister.repositores.components.ComponentProducerRepository;
import cz.osu.opr3.spaceshipregister.services.BaseService;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
public class ComponentProducerService extends BaseService<UUID, ComponentProducer, ComponentProducerRepository> {
    public ComponentProducerService(ComponentProducerRepository repository) {
        super(repository);
    }
    
    public ComponentProducer getByName(String name) {
        return this.repository.findByName(name).orElse(null);
    }
}

File: FE/Angular/src/app/services/models/entities/components/component-producer/component-producer.service.ts
import {Injectable} from '@angular/core';
import {ComponentProducer} from '../../../../../models/entities/components/component-producer';
import {BaseService} from '../../../base-service';

@Injectable({
              providedIn: 'root',
            })
export class ComponentProducerService extends BaseService<ComponentProducer> {
  constructor() {
    super('/component-producer');
  }
}
